highway.tex

```diff
@@ -212,73 +216,101 @@ \subsection{Votes and Weights}
+\ww(\{ \nu \in L_{p'} (\mu) \mid p(\nu) \}) & \geq \ww(S(\bar{J}(\lambda) \cap C'_{k-1}) \setminus E(\mu) \setminus B_k(\mu)) &\\
+& \geq \ww(\bar{J}(\lambda) \cap C'_{k-1}) - \ww(E(\mu) \cup B_k(\mu)) &\\
+& \geq \frac{1}{2} \ww(\mathcal{V}) + t - \ww(E(\mu)) -\ww(B_k(\mu) \setminus E(\mu)))&
+\end{flalign*}
+Together, these prove the inequality.
+\end{proof}
 
-Therefore, we have that $a+c\geq 2t(1-2^{-k+1}).$ Combining this with $a+2b\geq 2t$, we have that
-$$
-a+b+c \geq (2a+2b+c)/2 \geq (a+2b)/2+(a+c)/2 \geq 2t/2 + 2t(1-2^{-k+1})/2 = 2t(1-2^{-k})\text{.}
-$$
-The validators contributing to $a$, $b$ and $c$ are clearly disjoint. Those contributing to $a$ have equivocations in $J(\mu)$ and thus satisfy \ref{seen-equivocation}. Those contributing to $c$ have equivocations and $p'$-messages that are post-level-at-least-0 and contradict $p$, thus satisfying \ref{contradictory-equivocation}. For those contributing to $b$, we note that by the minimality of $\mu$, their vote for $\neg p$ could not have contained in its justification a message of level at least $k$, even the first message $\zeta$ of level at least $k$ that this validator must have cast. However, since this is their most recent $p'$-message it must be an equivocation against $\zeta$. Therefore, this equivocation would be contained in $\sigma \cup J(\mu)$. Furthermore, these validators will have cast post-level-0 $p'$-messages that contradict $p$, and hence they satisfy \ref{contradictory-equivocation}.
+\begin{theorem}
+Assume $p$ is $\ww$-majority-driven relative to $p'$. Let $(C_0, C_1, \ldots)$ be a summit, $\mu$ a $p'$-message with $\neg p(\mu)$ and $J(\mu) \cap C_k \neq \emptyset$. Let
+\begin{flalign*}
+B_l(\mu) &= \{S(\zeta) \mid \text{ $\zeta < \mu$, $p'(\zeta)$, $\neg p(\zeta)$, and there exists $C'_{l-1} \ni \nu' \sqsubset \zeta$ but no $C_l \ni \nu \sqsubset \zeta$.}\}\\
+A_l(\mu) &= B_l(\mu) \setminus E(\mu)\text{,}
+\end{flalign*}
+so $A_l(\mu)$ is the set of validators that $\mu$ sees voting against $p$ shortly after sending a $C'_{l-1}$ message, but not equivocating. $B_k(\mu)$ is as in Lemma \ref{SummitLemma}. Then:
+$$\ww(E(\mu)) + \sum_{l=1}^k \ww(A_l(\mu)) \;\geq\; 2t(1 - 2^{-k})$$
+In particular, at least $2t(1 - 2^{-k})$ validators (by weight) equivocated in any protocol state that contains both $C_0$ and $\mu$.
+\end{theorem}
 
-Thus, the validators contributing to $a,b$ and $c$ have total weight of at least $2t(1-2^{-k})$, and all satisfy either \ref{seen-equivocation} or \ref{contradictory-equivocation}.
+\begin{proof}
+If $\ww(E(\mu)) \geq 2t$, the inequality holds. So assume $\ww(E(\mu)) < 2t$.
+If $k = 0$, the right side is $0$, so the inequality is true.
+Assume now that $k > 0$ and that the theorem holds for $k - 1$.
+
+Note that all the $A_l(\mu)$ are disjoint:
+If a validator $v$ is in both $B_l(\mu)$ and $B_m(\mu)$ with $m > l$,
+then $\mu$ can see both a $\zeta$ by $v$ that cannot see any $C_l$-message by $v$,
+and a $\nu' \in C'_{k-1} \subseteq C_l$.
```

Comment:
I think the next sentences imply that we assume that `S(\nu')` is also `v` - is that the case? If yes, it might be nice to state it clearly here.